app/src/main/java/com/github/bgabriel998/softwaredevproject/ProfileActivity.java

```diff
@@ -92,22 +99,23 @@ public void submitUsernameButton(View view) {
         String username = ((EditText)findViewById(R.id.editTextUsername)).getText().toString();
         String currentUsername = account.getUsername();
         Log.d("CURRENT_USERNAME", "onSubmit: " + currentUsername);
-        // First, check if the username is valid
-        if(Account.isValid(username)) {
-            // Then, check if it is already used by the user
-            if(username.equals(currentUsername)) {
-                Log.d("CURRENT_USERNAME", "onSubmit: EQUAL_CASE");
-                Snackbar snackbar = Snackbar.make(findViewById(android.R.id.content), R.string.current_username, Snackbar.LENGTH_LONG);
-                snackbar.show();
-            }
-            // Finally, check if it is available
-            else Database.isPresent(Database.CHILD_USERS, Database.CHILD_USERNAME, username, () -> usernameAlreadyPresent(username), () -> registerUser(username));
-        }
-        // Display that the username is not valid
-        else {
-            Snackbar snackbar = Snackbar.make(findViewById(android.R.id.content), R.string.invalid_username , Snackbar.LENGTH_LONG);
+        // First, check if the username is valid or if it is already used by the user
+        if(!Account.isValid(username) || username.equals(currentUsername)) {
+            Snackbar snackbar = Snackbar.make(findViewById(android.R.id.content), !Account.isValid(username) ? R.string.invalid_username : R.string.current_username, Snackbar.LENGTH_LONG);
             snackbar.show();
         }
+        // Finally, check if it is available
+        else Database.isPresent("users", "username", username, () -> {
```

Comment:
Check if this suggestion works since the `CHILD_USERS` string contains `"/"`
```suggestion
        else Database.isPresent(Database.CHILD_USERS, Database.CHILD_USERNAME, username, () -> {
```

Reply:
Updated the paths in ProfileActivity.